library_generation/utilities.py

```diff
@@ -0,0 +1,69 @@
+
+from collections.abc import Sequence
+import sys
+
+
+
+"""
+Helper function for generate_composed_library.py
+
+It takes a single "query" argument
+"query" is a string of key=value entries separated by a comma
+example: "key1=value1,key2=value2,etc"
+
+it writes the same KVs in a format that generate_library.sh accepts to stdout
+example: "--key1 value1 --key2 value2"
+
+it ensures that both keys and values don't contain spaces
+"""
+def get_generate_library_arguments(query: str):
```

Comment:
nit: add a return type.
```suggestion
def get_generate_library_arguments(query: str) -> str:
```

Reply:
Good catch! I added them manually to all python functions